core/cat/memory/vector_memory.py

```diff
@@ -129,7 +132,28 @@ def create_collection_if_not_exists(self):
                 log(f'Collection "{self.collection_name}" has the same embedder', "INFO")
             else:
                 log(f'Collection "{self.collection_name}" has different embedder', "WARNING")
-                # TODO: dump collection on disk before deleting, so it can be recovered
+                # dump collection on disk before deleting, so it can be recovered
+                # TODO: it needs a code refactoring
+
+                #dormouse = "dormouse/"
+                if os.path.isdir("dormouse/"):
+                    log(f'Directory dormouse exists', "INFO")
+                else:
+                    os.mkdir("dormouse/")
+
+                self.snapshot_info = self.client.create_snapshot(collection_name=self.collection_name)
+                # TODO: parametrize cat host and port
+                snapshot_url_in = "http://cheshire_cat_vector_memory:6333/collections/" + self.collection_name + "/snapshots/"+ self.snapshot_info.name
```

Comment:
snapshot should only happen if Qdrant is on another address, and not in [`:memory:` mode](https://github.com/cheshire-cat-ai/core/blob/f579b4badd942bb6862a055fcebb959db788d3c5/core/tests/conftest.py#L27) or [file mode](https://github.com/cheshire-cat-ai/core/blob/f579b4badd942bb6862a055fcebb959db788d3c5/core/cat/memory/vector_memory.py#L68).

There should be a way to ask directly to `self.client`: if host and port are present as attributes, we can snapshot:
- http client is in fact created by giving host and port [here](https://github.com/cheshire-cat-ai/core/blob/f579b4badd942bb6862a055fcebb959db788d3c5/core/cat/memory/vector_memory.py#L88)
- `self.client` should contain that information [as attributes](https://github.com/qdrant/qdrant-client/blob/730ed8d239d79bc1adec0d7c2efc31b63be9b8e5/qdrant_client/qdrant_client.py#L72)

Reply:
Hi @pieroit I tried to print the attributes but

![immagine](https://github.com/cheshire-cat-ai/core/assets/11136646/2f2d4780-92c1-43ab-9bef-290bf2129c1c)

Reply:
Done in core/cat/memory/vector_memory.py from row 94 to 96